### PR TITLE
Storing Svrf access token in the keychain

### PR DESCRIPTION
**Pull request by AndreiEvstratenko**
# Pull Request

## Description

- storing access token in the KeyChain

Earlier we were storing our token and a token expiration time in the UserDefaults, but now our token is storing in the KeyChain, cause it's more secure. But, I have a question, do we need store our expiration time in the KeyChain or not? We have one difference between UserDefaults and KeyChain - if we delete an application from the device UserDefaults data will be removed, but KeyChain - won't. I was thinking about storing or not an expiration time and decided that would be better keep store this value in the UserDefaults, because it's not secure information and KeyChain is not a place for any data. 

### Motivation and Context

- security

### Types of changes

- [ ] Bug fix
- [ ] New feature
- [x] Maintenance

### CODE REVIEW

**Comment by AndreiEvstratenko** [comment]
@brentchow @ox pay attention on the question in the description, please

File: SvrfSDK/Source/SvrfSDK.swift [hunk 1]
```diff
@@ -358,12 +360,15 @@ public class SvrfSDK: NSObject {
          */
         private static func needUpdateToken() -> Bool {
 
-            if let tokenExpirationDate = UserDefaults.standard.object(forKey: svrfAuthTokenExpireDateKey) as? Date,
-                UserDefaults.standard.string(forKey: svrfAuthTokenKey) != nil {
+            if let tokenExpirationDate = UserDefaults.standard.object(forKey: svrfAuthTokenExpireDateKey) as? Date {
```
Comment (ox) [inline SvrfSDK/Source/SvrfSDK.swift]: I think you can reduce nesting here by doing a series of `guard` statements.

**Comment by brentchow** [comment]
Our jwt payload includes:

```
{
  "appName" "Foo":,
  "appId": "xxx",
  "appType": 5,
  "iat": 1547513112,
  "exp": 1552697112
}
```

**Comment by AndreiEvstratenko** [comment]
> Remove the nesting then you can merge this.
> 
> I don't think we need to store the expiration date in the keychain. Ideally we should have the `exp` value set on the JWT; not sure if we do. @brentchow do you know?

@ox, yes, our API gives us expiration time, but I need calculate the date and store this in the storage. So, I'll keep on store this in the UserDefaults.

**Comment by brentchow** [comment]
@AndreiEvstratenko I believe what @ox is saying is that we could parse the saved JWT to determine the expiration date by using the [`exp`](https://tools.ietf.org/html/rfc7519#section-4.1.4) value. If we do that, we won't have to save the `tokenExpirationDate`.

This is really smart and something we didn't do in the other apps. cc: @RomanFrom710

**Pull request merged**